text/0000-deprecate-http-tx.md

```diff
@@ -0,0 +1,109 @@
+
+For wallet developers, it is no longer necessary to be concerned about managing HTTP(S) endpoints. All transactions will be conducted via Tor or asynchronous exchange of armored slate strings or files. It can no longer be assumed that the counterparty in the transaction supports HTTP(S).
+
+Previously, some services like exchanges _only_ supported HTTP(S) Grin transactions. This change impacts these services significantly.
+
+Tor is the closest replacement to HTTP(S), but for some services the use of Tor is unacceptable. These services will will need to refactor the user flow to allow supporting armored slate strings instead, as the HTTP(S) method is no longer supported by default for `grin-wallet`.
+
+This up-front cost should be worth the reduced complexity required in supporting the alternative transaction methods as well as the privacy benefits provided to the users of the services.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+The technical and implementation details for this RFC are simple: no new features are added and some existing features are restricted:
+
+- `grin-wallet` http listener only accepts connections from localhost
+
+- `grin-wallet listen` no longer accepts `-e` argument
+
+- `grin-wallet send -d` provides a strong warning if the destination is not a valid .onion address
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+- Forces changes to existing transaction building workflows that use HTTP(S)
+
+- Remaining transaction options (Tor, armored slate, file) may not be appealing for services already exclusively using HTTP(S)
+
+- Forces "always on" wallet listeners to use Tor or to build out logic to support handling armored slates in the UX flow
+
+- Potentially makes a technology that is already challenging for services to adopt even more challenging to adopt
+
+- Those that do not value privacy as a first-order objective may see this as an unnecessary inconvenience
+
+- Friction could cause some services to avoid Grin completely rather than adding support for a new transaction method
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+Instead of looking for a best possible transaction method, this proposal comes down to favoring the least worst transaction methods by removing the worst transaction methods from the perspectives of privacy, security, censorship resistance and minimalism.
+
+In this case, HTTP(S) is the least desirable transaction method. It actively jeopardizes user privacy when services only accept payments via HTTP(S), can jeapardize security when not configured properly, is not censorship resistant as IPs can be blocked and domains can be seized and is not minimal once the additional configuration requirements are factored in.
```

Comment:
Generally not sure this belongs in this section. Be that as it may, I also have some comment on the content itself:

> is not censorship resistant as IPs can be blocked and domains can be seized

In that case, Tor is less censorship resistant as it's much easier to blanket ban that method, whereas with https you'd have to block individual IPs. And while you can seize individual domains, you can also seize Tor services, so not sure this is a valid motivation.
 
> and is not minimal once the additional configuration requirements are factored in.

But then Tor is not minimal either.

Reply:
Removed this, it did not fit